python/postprocessing/modules/jme/jetSmearer.py

```diff
@@ -17,7 +17,12 @@ def __init__(self, globalTag, jetType = "AK4PFchs", jerInputFileName = "Spring16
 
         # read jet energy resolution (JER) and JER scale factors and uncertainties
         # (the txt files were downloaded from https://github.com/cms-jet/JRDatabase/tree/master/textFiles/Spring16_25nsV10_MC )
-        self.jerInputFilePath = os.environ['CMSSW_BASE'] + "/src/PhysicsTools/NanoAODTools/data/jme/"
+        # Text files are now tarred so must extract first
+        self.jerInputArchivePath = os.environ['CMSSW_BASE'] + "/src/PhysicsTools/NanoAODTools/data/jme/"
+        self.jerTag = jerInputFileName[:jerInputFileName.find('_MC_')+len('_MC')]
+        self.jerArchive = tarfile.open(self.jerInputArchivePath+self.jerTag+".tgz", "r:gz")
+        self.jerInputFilePath = "scratch/"+self.jerTag
```

Comment:
we should rather create a unique name with some mktemp function to avoid clashes

Reply:
I did not know mktemp was a thing! It makes sense of course, I just never thought to look for a dedicated tool :-) I've been playing with the built-in version for python from `tempfile` and it works as intended in my minimum working example. It doesn't look like we need to remove the directory ourselves (it gets deleted once python cleans up the object) so I'll remove the call to shutil as well. Will commit ASAP.